### ISSUE TITLE
Completions broken for pre-qualified import

### ISSUE THREAD

**Issue by Jashweii** [haskell/haskell-language-server#2824]
```Haskell
import qualified Control.Monad as M(|)
                                    ^ cursor
```
Doesn't suggest imports from Control.Monad, unlike post-qualified and unqualified imports. 
I don't think it matters if "as" is present or not.

I get this on both: 
HLS current master / GHC 9.2.2 / WSL Ubuntu / vscode (WSL)
HLS 1.6.1.0 / GHC 9.2.1 / Windows / vscode

**Comment by xsebek** [comment]
Hi @Jashweii, thanks for the bug report. :slightly_smiling_face: 

The issue seems to be with `ghcide` which provides the completion.

I wrote a test case in PR #2834, but I have not yet figured out where the problem is.

**Comment by xsebek** [comment]
Mystery resolved!
```haskell
-- IDE/Plugin/Completions/Logic.hs
-- getCompletions
let moduleName = T.pack $ words (T.unpack fullLine) !! 1
```

**Comment by Jashweii** [comment]
This is fixed for me as of hls-2.6.0.0 on ghc-9.8.1
